City-of-Helsinki/events-helsinki-monorepo
# LIIKUNTA-655: Add dynamic CMS configured redirect to Liikunta only

## Description

### feat(sports): add dynamic CMS configured redirect using middleware

The redirects are fetched periodically from CMS's GraphQL endpoint for
siteSettings.redirects. The redirects' cache is in-memory and per pod,
and the cache is refreshed from CMS by default between 6–9 minutes
(i.e. 5–10min interval but a bit tighter and under 10min).

Redirects' cache updating interval can be configured using environment
variables:
 - MIN_REDIRECTS_UPDATE_INTERVAL_MS (defaults to 6min in milliseconds)
 - MAX_REDIRECTS_UPDATE_INTERVAL_MS (defaults to 9min in millseconds)

Also update generated GraphQL schemas i.e. ran:
```bash
cd apps/sports-helsinki/
yarn generate:graphql
cp ./src/components/domain/graphql/generated/graphql.tsx \
   ../../packages/components/src/types/generated/
cd ../../
npx prettier ./packages/components/src/types/generated/graphql.tsx -w
rm -fr ./apps/sports-helsinki/src/components/
```
and as a consequence disambiguate conflicting PageInfo types,
PageInfo, PageUriInfo and VenueProxyPageInfo.

refs LIIKUNTA-655

## Issues

### Closes

[LIIKUNTA-655](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-655)

### Related

## Testing

### Automated tests

### Manual testing

- Login into liikunta test CMS at https://liikunta.app-staging.hkih.hion.dev/wp-login.php
- Go to "Sivuston asetukset" (i.e. Site settings) / Redirects and set & save the wanted redirects
  - e.g. ![image](https://github.com/user-attachments/assets/c59768e9-631d-4ccd-89b8-5cb0e4aff185)
- Wait for 10 minutes
- Now the redirects should be in effect in the PR's liikunta environment i.e. https://liikunta-pr726.dev.hel.ninja/

## Screenshots

## Additional notes


[LIIKUNTA-655]: https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
